pkg/kubectl/cmd/util/helpers.go

```diff
@@ -400,3 +401,21 @@ func UpdateObject(info *resource.Info, updateFn func(runtime.Object) error) (run
 	}
 	return info.Object, nil
 }
+
+type groupID int
+
+const (
+	unknownGroup groupID = iota
+	apiGroup
+	experimentalGroup
+)
+
+func groupForResource(resource string) groupID {
```

Comment:
Then we'd have to explicitly tell the RESTMapper what group it is for. While I'm not against that, I would rather defer something like that to proper multi-group support. I'm also not happy with the way the client api works (and I think others feel the same), so I don't see a clean solution. If you still feel that's the right approach then I'm willing to change it.